frontend/projects/ui/src/app/pages/apps-routes/app-show/pipes/to-buttons.pipe.ts

```diff
@@ -112,23 +115,67 @@ export class ToButtonsPipe implements PipeTransform {
     await modal.present()
   }
 
-  private viewInMarketplaceButton(pkg: PackageDataEntry): Button {
-    return pkg.installed?.['marketplace-url']
-      ? {
-          action: () =>
-            this.navCtrl.navigateForward([`marketplace/${pkg.manifest.id}`]),
-          title: 'Marketplace',
-          description: 'View service in marketplace',
-          icon: 'storefront-outline',
-        }
-      : {
-          disabled: true,
-          action: () => {},
-          title: 'Marketplace',
-          description:
-            'This package has been side-loaded and is not available in the Start9 Marketplace',
-          icon: 'storefront-outline',
-        }
+  private viewInMarketplaceButton(
+    pkg: PackageDataEntry,
+    currentMarketplace: Marketplace | null,
+  ): Button {
+    const pkgMarketplace = pkg.installed?.['marketplace-url']
+    const pkgM = pkgMarketplace
+      ? removeTrailingSlash(pkgMarketplace)
+      : pkgMarketplace
+    const currentM = currentMarketplace
+      ? removeTrailingSlash(currentMarketplace.url)
+      : currentMarketplace
+    if (!pkgMarketplace) {
```

Comment:
Should probably start with this conditional, then you don't have to use the ternaries above.